Data-Structures-and-Algos-Topic/Module1-Intro-to-Data-Structures-and-Algos/Activity2_HashTables/cards/9.md

```diff
@@ -4,38 +4,83 @@
 
  **Example Output**:
 
- Input: "abcabcbb"
- Output: 3 
- Explanation: The answer is "abc", with the length of 3. 
+ Input: "ABDEFGABEF"
```

Comment:
I haven't seen the previous cards yet, but the explanation of the solution (lines 14-19) are not very clearly explained. It glosses over a lot of the why's. consider:
- why do we store the indices of the last visited characters in addition to the length? is it important to keep track of when we eventually return the answer?
- you never explicitly mention that you are essentially breaking the string into 2 parts (visited and not-visited). this can be confusing
- what is the purpose of the sentence "Let the maximum length be max_len"? is this some variable that we first set when we start? will it be continually updated to store the new maximum NRCSs?